lib/mobility-core/src/Kernel/External/AadhaarVerification/Gridline/Types.hs

```diff
@@ -0,0 +1,115 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+{-# LANGUAGE DerivingStrategies #-}
+
+module Kernel.External.AadhaarVerification.Gridline.Types where
+
+import Data.Aeson
+import EulerHS.Prelude hiding (error, state, (.=))
+
+data GridlineAadhaarOtpReq = GridlineAadhaarOtpReq
+  { aadhaar_number :: Text,
+    consent :: Text
+  }
+  deriving (Generic, FromJSON, ToJSON, Show)
+
+data GridlineVerifyAadhaarData = GridlineVerifyAadhaarData
+  { code :: Text,
+    message :: Text,
+    transaction_id :: Text
+  }
+  deriving (Show, Generic, ToJSON, FromJSON)
+
+data GridlineVerifyAadhaarError = GridlineVerifyAadhaarError
+  { code :: Text,
+    message :: Text
+  }
+  deriving (Show, Generic, ToJSON, FromJSON)
+
+data GridlineVerifyAadhaarResp = GridlineVerifyAadhaarResp
+  { request_id :: Text,
+    status :: Int,
+    _data :: Maybe GridlineVerifyAadhaarData,
+    _error :: Maybe GridlineVerifyAadhaarError,
+    path :: Text
+  }
+  deriving (Show, Generic, ToJSON)
+
+instance FromJSON GridlineVerifyAadhaarResp where
+  parseJSON = withObject "GridlineVerifyAadhaarResp" $ \v -> do
+    request_id <- v .: "request_id"
+    path <- v .: "path"
+    status <- v .: "status"
+    _data <- v .:? "data"
+    _error <- v .:? "error"
+    return (GridlineVerifyAadhaarResp request_id status _data _error path)
```

Comment:
```
instance FromJSON GridlineVerifyAadhaarResp where
  parseJSON = genericParseJSON stripPrefixUnderscoreIfAny

instance ToJSON GridlineVerifyAadhaarResp where
  toJSON = genericToJSON stripPrefixUnderscoreIfAny
```

Reply:
there is field called data so how will i handle that

Reply:
`stripPrefixUnderscoreIfAny` used for this. On Haskell level you'll have `_data`, but on JSON level `data`